src/XliffTasks/Model/Document.cs

```diff
@@ -62,12 +62,37 @@ public void Load(Stream stream)
         /// </summary>
         public void Save(string path)
         {
+            //On Windows:
+            // Readers will prevent the file from being overwritten due to FileShare.Read.
+            // Readers can read in parallel, but when there's contention with a writer, the retrying will kick in to resolve it.
+            //On Unix:
+            // FileShare.Read does nothing, but...
+            // File.Replace is implemented with rename system call that will mean that even though writers can overwrite while 
+            // reading is happening, each reader will see file before or after overwrite, not in between
+
             EnsureContent();
+            var tempPath = Path.Combine(Path.GetDirectoryName(path), Path.GetRandomFileName());
 
-            using (var stream = File.Open(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+            using (var stream = File.Open(tempPath, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
             {
                 Save(stream);
             }
+
+            ExponentialRetry.ExecuteWithRetryIOException(() =>
```

Comment:
I'd add the word on between retry and IO to the method name

Reply:
ExecuteWithRetryOnIOException

Reply:
fixed